source/virtualBuffers/gecko_ia2.py

```diff
@@ -576,3 +631,106 @@ def _getInitialCaretPos(self):
 		if initialPos:
 			return initialPos
 		return self._initialScrollObj
+
+	# C901 'updateAppSelection' is too complex
+	# Note: when working on updateAppSelection, look for opportunities to simplify
+	# and move logic out into smaller helper functions.
+	def updateAppSelection(self):  # noqa: C901

```

Comment:
can this function be simplified? I think we should avoid introducing complex functions where possible

Reply:
Unfortunately I don't think I have the time that this would require at the moment. You are welcome to have a go if you believe this is necessary for this pr to be merged.